sup3r/bias/bias_transforms.py

```diff
@@ -520,3 +525,206 @@ def local_qdm_bc(data: np.array,
+        Nearest neighbor euclidean distance threshold. If the coordinates are
+        more than this value away from the bias correction lat/lon, an error
+        is raised.
+
+    Returns
+    -------
+    base : np.array
+        Parameters used to define the statistical distribution estimated for
+        the ``base_dset``. It has a shape of (I, J, P), where (I, J) are the
+        same first two dimensions of the given `lat_lon` and P is the number
+        of parameters and depends on the type of distribution. See
+        :class:`~sup3r.bias.bias_calc.QuantileDeltaMappingCorrection` for more
+        details.
+    bias : np.array
+        Parameters used to define the statistical distribution estimated for
+        (historical) ``feature_name``. It has a shape of (I, J, P), where
+        (I, J) are the same first two dimensions of the given `lat_lon` and P
+        is the number of parameters and depends on the type of distribution.
+        See :class:`~sup3r.bias.bias_calc.QuantileDeltaMappingCorrection` for
+        more details.
+    bias_fut : np.array
+        Parameters used to define the statistical distribution estimated for
+        (future) ``feature_name``. It has a shape of (I, J, P), where (I, J)
+        are the same first two dimensions of the given `lat_lon` and P is the
+        number of parameters used and depends on the type of distribution. See
+        :class:`~sup3r.bias.bias_calc.QuantileDeltaMappingCorrection` for more
+        details.
+    cfg : dict
+        Metadata used to guide how to use of the previous parameters on
+        reconstructing the statistical distributions. For instance,
+        `cfg['dist']` defines the type of distribution. See
+        :class:`~sup3r.bias.bias_calc.QuantileDeltaMappingCorrection` for more
+        details, including which metadata is saved.
+
+    Warnings
+    --------
+    Be careful selecting which `bias_fp` to use. In particular, if
+    "bias_fut_{feature_name}_params" is representative for the desired target
+    period.
+
+    See Also
+    --------
+    sup3r.bias.bias_calc.QuantileDeltaMappingCorrection
+        Estimate the statistical distributions loaded here.
+
+    Examples
+    --------
+    >>> lat_lon = np.array([
+    ...              [39.649033, -105.46875 ],
+    ...              [39.649033, -104.765625]])
+    >>> params, cfg = get_spatial_bc_quantiles(
+    ...                 lat_lon, "ghi", "rsds", "./dist_params.hdf")
+    """
+    ds = {'base': f'base_{base_dset}_params',
+          'base_zero_rate': f'{base_dset}_zero_rate',
+          'bias': f'bias_{feature_name}_params',
+          'bias_fut': f'bias_fut_{feature_name}_params',
+          'bias_mean_mh': f'{feature_name}_mean_mh',
+          'bias_mean_mf': f'{feature_name}_mean_mf',
+          }
+    params = _get_factors(lat_lon, ds, bias_fp, threshold)
+
+    with Resource(bias_fp) as res:
+        cfg = res.global_attrs
+
+    return params, cfg
+
+
+def local_presrat_bc(data: np.array,
```

Comment:
doc string here. also `np.ndarray` is the correct type hint, I believe.

Reply:
Opps! Good catch! Thanks.